lua/octo/commands.lua

```diff
@@ -1117,43 +1117,32 @@ function M.pr_checks()
 end
 
 function M.merge_pr(...)
-  local conf = config.values
-  local defaultMergeMethod = conf.default_merge_method
-
   local bufnr = vim.api.nvim_get_current_buf()
   local buffer = octo_buffers[bufnr]
   if not buffer or not buffer:isPullRequest() then
     return
   end
+
   local args = { "pr", "merge", tostring(buffer.number) }
   local params = table.pack(...)
-  for i = 1, params.n do
-    if params[i] == "delete" then
-      table.insert(args, "--delete-branch")
-    end
-  end
-  local has_flag = false
-  for i = 1, params.n do
-    if params[i] == "commit" then
-      table.insert(args, "--merge")
-      has_flag = true
-    elseif params[i] == "squash" then
-      table.insert(args, "--squash")
-      has_flag = true
-    elseif params[i] == "rebase" then
-      table.insert(args, "--rebase")
-      has_flag = true
+  local conf = config.values
+
+  local merge_method = conf.default_merge_method
+  for _, param in ipairs(params) do
+    if utils.merge_method_to_flag[param] then
+      merge_method = param
     end
   end
-  if not has_flag then
-    if defaultMergeMethod == "squash" then
-      table.insert(args, "--squash")
-    elseif defaultMergeMethod == "rebase" then
-      table.insert(args, "--rebase")
-    else
-      table.insert(args, "--merge")
+  utils.insert_merge_flag(args, merge_method)
+
+  local delete_branch = conf.default_delete_branch
+  for _, param in ipairs(params) do
+    if param == "delete" then
+      delete_branch = true
```

Comment:
we can also add `nodelete` param to `Octo pr merge` to turn it off if `default_delete_branch=true`

Reply:
added `nodelete` argument to `Octo pr merge`

Reply:
Thanks!